src/passes/OptimizeInstructions.cpp

```diff
@@ -1127,17 +1132,19 @@ struct OptimizeInstructions
     ZeroRemover remover(getPassOptions());
     remover.setModule(getModule());
     remover.walk(walked);
-    if (constant == 0) {
+    if (constant == 0ULL) {
       return walked; // nothing more to do
     }
     if (auto* c = walked->dynCast<Const>()) {
-      assert(c->value.geti32() == 0);
-      c->value = Literal(constant);
+      assert(c->value.getInteger() == 0LL);
+      c->value = Literal::makeFromInt64(constant, c->type);
```

Comment:
This may need to be an int32 in some cases?

Reply:
But I'm not sure about this. Not all 64-bit values fit in an i32. Is the assumption that it's fine because of wrapping? Then a comment would be good.

Reply:
I'll add comment. Thanks for review!